### PR TITLE
New Basic MGH Reader and Viewer Functionality for MRI Data

### PR DESCRIPTION
**Pull request by CBoensel**
ported functionality from freesurfer

### CODE REVIEW

**Comment by CBoensel** [comment]
any comments? which are the most important improvements needed?

File: MNE/fs/mgh.cpp [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+//=============================================================================================================
+
+Mgh::Mgh()
+{
+
+}
+
+//*************************************************************************************************************
+
+//QList<MatrixXd>
```
Comment (LorenzE) [inline MNE/fs/mgh.cpp]: delete this

File: MNE/fs/mgh.cpp [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+    mri.fName = fName;
+
+    mri.slices = sliceDataList;
+    return mri;
+//    return sliceDataList;
+}
+
+//*************************************************************************************************************
+
+// function to unpack compressed data
```
Comment (LorenzE) [inline MNE/fs/mgh.cpp]: function documentation are to be made in the header file

File: MNE/fs/mri.cpp [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+//=============================================================================================================
+
+Mri::Mri()
+{
+
+}
+
+//*************************************************************************************************************
+
+// width, height, depth, type
```
Comment (LorenzE) [inline MNE/fs/mri.cpp]: move this to header documentation

File: MNE/fs/slicedata.h [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+namespace FSLIB
+{
+
+//*************************************************************************************************************
+//=============================================================================================================
+// USED NAMESPACES
+//=============================================================================================================
+
+using namespace Eigen;
+
```
Comment (LorenzE) [inline MNE/fs/slicedata.h]: doxygen!

**Comment by LorenzE** [comment]
Also, please try to fix the data index error when reading the mri slice data. This is very important!

**Comment by CBoensel** [comment]
please precise the last statement. don't see any index errors on my machine.

**Comment by LorenzE** [comment]
My post was regarding that the final MRI slice is shifted in the view. There are no actual errors produced during compilation, but the final (visualised) result is not correct. The view is showing a shifted/half version of the slice. You showed me this error in our weekly meeting two weeks ago. Our assumption was that the start index for reading the data from the 1D data array is not chosen correctly.

**Comment by CBoensel** [comment]
understood! thx for the refinement.

**Pull request closed without merging**